universal_doc_importer/file_system.py

```diff
@@ -38,9 +38,10 @@ def printAllChildren(self, depth=-1):
             print("\t" * depth + "}")
 
     def makeDict(self):
-        dictionary = {self.name: []}
         if len(self.children) > 0:
+            dictionary = {self.name: []}
             for child in self.children:
                 dictionary[self.name].append(child.makeDict())
-
-        return dictionary
+            return dictionary
+        else:
+            return self.name
```

Comment:
Could you please revert my changes back, because returning only the name causes an error in some cases. It's better to make it more robust.

Reply:
And let's remove this change at all, because it's not related to the issue's theme